geode-core/src/main/java/org/apache/geode/cache/client/internal/PoolImpl.java

```diff
@@ -136,6 +136,7 @@
   private final OpExecutorImpl executor;
   private final RegisterInterestTracker riTracker = new RegisterInterestTracker();
   private final InternalDistributedSystem dsys;
+  private final Cache cache;
```

Comment:
I think you should make this InternalCache instead of Cache since that is what you have when you pass it in and it is used internally